pkg/ui/workspaces/cluster-ui/src/databases/util.tsx

```diff
@@ -240,34 +241,49 @@ export function buildIndexStatToRecommendationsMap(
   return recommendationsMap;
 }
 
-export function checkInfoAvailable(
-  requestError: Error,
-  queryError: Error,
-  cell: React.ReactNode,
-): React.ReactNode {
+interface LoadingCellProps {
+  requestError: Error;
+  queryError?: Error;
+  loading: boolean;
+}
+
+export const LoadingCell: React.FunctionComponent<LoadingCellProps> = ({
+  loading,
+  requestError,
+  queryError,
+  children,
+}) => {
+  if (loading) {
+    return (
+      <Skeleton loading={true} active={true} paragraph={false} title={true} />
+    );
+  }
+
   let tooltipMsg = "";
   if (requestError) {
     tooltipMsg = `Encountered a network error fetching data for this cell: ${requestError.name}`;
   } else if (queryError) {
     tooltipMsg = getQueryErrorMessage(queryError);
-  } else if (cell == null) {
-    tooltipMsg = "Empty result";
```

Comment:
I see that previously and now we show either the error tooltip or the cell but never both.
I wonder what happens right now if a query hits the 'max size exceeded' error but still returns rows (e.g. the query fetching tables for a db). I recall this is something the original page was supposed to do so I wonder if there is other logic this process that deals with this right now but it might be worth testing before we backport/merge.